### DISCUSSION TITLE
RD4J META-INF/services manually included in apk file

### DISCUSSION THREAD

**Discussion by Antus8** [eclipse-rdf4j/rdf4j#3554] [Q&A]
I'm trying to develop an application for App Inventor using an extension that relies on rdf4j. The problem is that when the apk file is generated the META-INF/services are not included and the application doesn't work.
Since I'm not using Android Studio or some other development environment I'm wondering if I can add manually the META-INF/services required by rdf4j in the apk file and where I have to place them to make the system work. 
Any solution? Thanks in advance

**Comment by hmottestad** [comment]
I'm unfortunately not versed in android development. Could you elaborate a bit on your use case and which version of RDF4J you are using.

**Reply by hmottestad** [reply]
Found this: https://ibotpeaches.github.io/Apktool/documentation/

There is a command called `--copy-original`, maybe that could help? Also there is some discussion there about unsigned apk files not having a meta-inf folder, so maybe you need to sign your build? Honestly don't know much about this.
